From: Benas Bagvilas <benas.bagvilas@teltonika.lt>
Date: Mon Nov 4 11:14:30 2024 +0200
Subject: crypto: mtk-eip93: fix double return on check_valid_request fail

Error branch for check_valid_request calls return the error to the crypto API user both 
synchronously via function return and asynchronously via complete callback, resulting in
possible double frees and other issues in user code.

Signed-off-by: Benas Bagvilas <benas.bagvilas@teltonika.lt>
---
 drivers/crypto/mtk-eip93/eip93-aead.c   |    1 -
 drivers/crypto/mtk-eip93/eip93-cipher.c |    1 -
 2 files changed, 2 deletions(-)

--- a/drivers/crypto/mtk-eip93/eip93-aead.c
+++ b/drivers/crypto/mtk-eip93/eip93-aead.c
@@ -36,7 +36,6 @@ static int mtk_aead_send_req(struct mtk_
 
 	err = check_valid_request(rctx);
 	if (err) {
-		aead_request_complete(req, err);
 		return err;
 	}
 
--- a/drivers/crypto/mtk-eip93/eip93-cipher.c
+++ b/drivers/crypto/mtk-eip93/eip93-cipher.c
@@ -34,7 +34,6 @@ static int mtk_skcipher_send_req(struct
 	err = check_valid_request(rctx);
 
 	if (err) {
-		skcipher_request_complete(req, err);
 		return err;
 	}
 
